palette/src/hsluv.rs

```diff
@@ -0,0 +1,689 @@
+where
+    T: FloatComponent,
+    Wp: WhitePoint,
+{
+    fn clone(&self) -> Hsluv<Wp, T> {
+        *self
+    }
+}
+
+impl<T> Hsluv<D65, T>
+where
+    T: FloatComponent,
+{
+    /// HSLuv with default D65 white point.
+    pub fn new<H: Into<LabHue<T>>>(hue: H, saturation: T, l: T) -> Self {
+        Hsluv {
+            hue: hue.into(),
+            saturation,
+            l,
+            white_point: PhantomData,
+        }
+    }
+}
+impl<Wp, T> Hsluv<Wp, T>
+where
+    T: FloatComponent,
+    Wp: WhitePoint,
+{
+    /// HSLuv with sepcified D65 white point.
+    pub fn with_wp<H: Into<LabHue<T>>>(hue: H, saturation: T, l: T) -> Self {
+        Hsluv {
+            hue: hue.into(),
+            saturation,
+            l,
+            white_point: PhantomData,
+        }
+    }
+
+    /// Convert to a `(hue, saturation, L\*)` tuple.
+    pub fn into_components(self) -> (LabHue<T>, T, T) {
+        (self.hue, self.saturation, self.l)
+    }
+
+    /// Convert from a `(hue, saturation, L\*)` tuple.
+    pub fn from_components<H: Into<LabHue<T>>>((hue, saturation, l): (H, T, T)) -> Self {
+        Self::with_wp(hue, saturation, l)
+    }
+
+    /// Return the `l` value minimum.
+    pub fn min_l() -> T {
+        T::zero()
+    }
+
+    /// Return the `l` value maximum.
+    pub fn max_l() -> T {
+        from_f64(100.0)
+    }
+
+    /// Return the minimum `saturation` value.
+    pub fn min_saturation() -> T {
+        T::zero()
+    }
+    /// Return the maximum `saturation` value.
+    pub fn max_saturation() -> T {
+        from_f64(100.0)
+    }
+
+    /// Return sum of the absolute difference of components.
+    fn abs_diff(&self, other: &Self) -> T {
+	(self.hue - other.hue).to_positive_degrees() + (self.saturation - other.saturation).abs() +
+	    (self.l - other.l).abs()
+    }
+
+    fn max_component(&self) -> T {
+	let h = self.hue.to_positive_degrees();
+	if h > self.l {
+	    if h > self.saturation {
+		h
+	    } else {
+		self.saturation
+	    }
+	} else if self.l > self.saturation {
+	    self.l
+	} else {
+	    self.saturation
+	}
+
+    }
+
+    /// Return the maximum `chroma` value that is representable for
+    /// the given lightness and hue values.
+    pub fn max_chroma_for_lh(&self) -> T {
+        LuvBounds::from_lightness(self.l).max_chroma_at_hue(self.hue)
+    }
+}
+
+impl<Wp, T> PartialEq for Hsluv<Wp, T>
+where
+    T: FloatComponent + PartialEq,
+    Wp: WhitePoint,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.hue == other.hue && self.saturation == other.saturation && self.l == other.l
+    }
+}
+
+impl<Wp, T> Eq for Hsluv<Wp, T>
+where
+    T: FloatComponent + Eq,
+    Wp: WhitePoint,
+{
+}
+
+///<span id="Hsluva"></span>[`Hsluva`](crate::Hsluva) implementations.
+impl<T, A> Alpha<Hsluv<D65, T>, A>
+where
+    T: FloatComponent,
+    A: Component,
+{
+    /// CIE L\*C\*h° and transparency with white point D65.
+    pub fn new<H: Into<LabHue<T>>>(hue: H, saturation: T, l: T, alpha: A) -> Self {
```

Comment:
HSLuv is supposed to be based on Luv! I think what you are making here is HSLab instead.

Reply:
While correct, it seems like your comment got misplaced.